### PR TITLE
Updating LRO documentation to reflect new design

### PR DESCRIPTION
**Pull request by catalinaperalta**

### CODE REVIEW

File: docs/golang/design.md [hunk 1]
```diff
@@ -340,33 +340,49 @@ type WidgetPoller interface {
 	// If Poll succeeds and the operation has completed successfully, the WidgetPoller
 	// is updated and the Widget is returned.
 	// If Poll succeeds and the operation has not completed, the WidgetPoller is
-	// updated and both return values are nil.
-	Poll(context.Context) (*WidgetResponse, error)
+	// updated and returns the latest http response.
```
Comment (jhendrixMSFT) [inline docs/golang/design.md:343]: HTTP

File: docs/golang/design.md [hunk 2]
```diff
@@ -378,47 +394,70 @@ func (c *WidgetClient) ResumeWidgetPoller(id string, reporter func(string)) Widg
 {% include requirement/MUST id="golang-lro-pattern" %} follow the operation pattern for all LROs.
 
 ```go
-// example #1, blocking call to Wait()
+// example #1, blocking call to PollUntilDone()
 options := BeginCreateOptions{
```
Comment (jhendrixMSFT) [inline docs/golang/design.md:398]: I believe we decided to remove the `Reporter` yes?

File: docs/golang/design.md [hunk 3]
```diff
@@ -378,47 +394,70 @@ func (c *WidgetClient) ResumeWidgetPoller(id string, reporter func(string)) Widg
...
+	// handle error ...
+}
+process(w)
 
 // example #3, resuming from a previous operation
-poller := client.ResumeWidgetPoller("operation ID from external process", func(msg string){
-	fmt.Printf("wiget creation progress: %s", msg)
-})
-w, err := poller.Wait(context.Background(), 2*time.Second)
+poller, err := client.ResumeWidgetPoller("resume token from external process")
```
Comment (jhendrixMSFT) [inline docs/golang/design.md:440]: Can you please update this to how you do it in the tests?  i.e. show the call to get a resume token then pass it here.  You could still keep the comment about the resume token can come from a different process.

File: docs/golang/design.md [hunk 4]
```diff
@@ -378,47 +394,70 @@ func (c *WidgetClient) ResumeWidgetPoller(id string, reporter func(string)) Widg
...
 if err != nil {
-	// handle error...
+	// handle error ...
+}
+for {
+	resp, err := p.Poll(context.Background())
```
Comment (jhendrixMSFT) [inline docs/golang/design.md:445]: Hmm one thing I just realized is that with the `WaitUntilDone()` method on the response envelope, we lose it when hydrating a resume token.  We should discuss next week when Jeff is back (maybe we don't really care too much, it's an advanced scenario).

Comment (catalinaperalta) [reply docs/golang/design.md:445]: Yes I just noticed that as well. Agreed, let's discuss next week.

File: docs/golang/design.md (hunk 2 above)
Comment (catalinaperalta) [reply docs/golang/design.md:398]: Yes removing it

File: docs/golang/design.md (hunk 1 above)
Comment (catalinaperalta) [reply docs/golang/design.md]: Done

File: docs/golang/design.md (hunk 2 above)
Comment (catalinaperalta) [reply docs/golang/design.md]: Done

File: docs/golang/design.md (hunk 3 above)
Comment (catalinaperalta) [reply docs/golang/design.md]: Done

File: docs/golang/design.md [hunk 5]
```diff
@@ -340,85 +340,124 @@ type WidgetPoller interface {
...
+}
+
 // BeginCreate creates a new widget with the specified name.
-func (c *WidgetClient) BeginCreate(ctx context.Context, name string, options *BeginCreateOptions) (WidgetPoller, error) {
+func (c *WidgetClient) BeginCreate(ctx context.Context, name string, options *BeginCreateOptions) (*WidgetResponse, error) {
 	// ...
 }
 ```
 
 {% include requirement/MUST id="golang-lro-resuming-operations" %} provide a method with the prefix `Resume` to instantiate a `<Resource>Poller` type with the `ID` from a previous call to `Operation.ID()`.
```
Comment (jhendrixMSFT) [inline docs/golang/design.md:380]: I think this text also needs to be updated to replace ID with resume token etc

File: docs/golang/design.md [hunk 6]
```diff
@@ -340,85 +340,124 @@ type WidgetPoller interface {
...
+process(w)
 
 // example #3, resuming from a previous operation
-poller := client.ResumeWidgetPoller("operation ID from external process", func(msg string){
-	fmt.Printf("wiget creation progress: %s", msg)
-})
-w, err := poller.Wait(context.Background(), 2*time.Second)
+// getting the resume token from a previous poller instance
+poller := resp.Poller
+tk := poller.ResumeToken()
```
Comment (jhendrixMSFT) [inline docs/golang/design.md:435]: `tk, err`

Comment (catalinaperalta) [reply docs/golang/design.md]: Done

File: docs/golang/design.md (hunk 5 above)
Comment (catalinaperalta) [reply docs/golang/design.md]: Done

File: docs/golang/design.md [hunk 7]
```diff
@@ -340,85 +340,124 @@ type WidgetPoller interface {
...
-poller, err := client.BeginCreate(context.Background(), "green_widget")
+// example #2, customized poll loop
+resp, err := client.BeginCreate(context.Background(), "green_widget")
 if err != nil {
 	// handle error...
 }
+poller := resp.Poller
 for {
-	w, err := poller.Poll(context.Background())
+	resp, err := p.Poll(context.Background())
```
Comment (jhendrixMSFT) [inline docs/golang/design.md:413]: The var is named `poller` but here we use `p` (there are a few other cases where this happens).

**Pull request merged**